preprocessors/object-detection-azure/objdetect.py

```diff
@@ -0,0 +1,194 @@
+# load the schema
+with open('./schemas/preprocessors/object-detection.schema.json') \
+        as jsonfile:
+    data_schema = json.load(jsonfile)
+with open('./schemas/preprocessor-response.schema.json') \
+        as jsonfile:
+    schema = json.load(jsonfile)
+with open('./schemas/definitions.json') as jsonfile:
+    definitionSchema = json.load(jsonfile)
+with open('./schemas/request.schema.json') as jsonfile:
+    first_schema = json.load(jsonfile)
+
+
+def process_results(result):
+    send = []
+    image_height = result["metadata"]["height"]
+    image_width = result["metadata"]["width"]
+    for item in range(len(result['objects'])):
+        dims = result["objects"][item]['rectangle']
+        x_left = dims['x'] / image_width
+        x_right = (dims['x'] + dims['w']) / image_width
+        y_left = dims['y'] / image_height
+        y_right = (dims['y'] + dims['h']) / image_height
+        centre = [abs((x_left + x_right) / 2),
+                  abs((y_left + y_right) / 2)]
+        area = abs((dims['w'] * dims['h']) / (image_height * image_width))
+        class_prob = float(result['objects'][item]['confidence'])
+        normalised_dims = [x_left, y_left, x_right, y_right]
+        dictionary = {"ID": item,
+                      "type": result['objects'][item]['object'],
+                      "dimensions": normalised_dims,
+                      "confidence": class_prob,
+                      "centroid": centre,
+                      "area": area
+                      }
+        send.append(dictionary)
+    return send
+
+
+def process_image(image):
+
+    region = "canadacentral"  # For example, "westus"
+
+    try:
+        api_key = os.environ["AZURE_API_KEY"]
+    except Exception as e:
+        logging.error(e)
+        return "", 500
+
+    # Set request headers
+    headers = dict()
+    headers['Ocp-Apim-Subscription-Key'] = api_key
+    headers['Content-Type'] = 'application/octet-stream'
+
+    # Set request querystring parameters
+    """Only query categories"""
+    params = {'visualFeatures': 'Objects'}
+
+    # Make request and process response
+    add = "https://{}.api.cognitive.microsoft.com/vision/v1.0/analyze"
+    response = requests.request(
+        'post',
+        add.format(region),
+        data=image,
+        headers=headers,
+        params=params
+    )
+    if response.status_code == 200 or response.status_code == 201:
+
+        if 'content-length' in response.headers and \
+                int(response.headers['content-length']) == 0:
+            return "Invalid response from azure"
+        elif 'content-type' in response.headers and \
+                isinstance(response.headers['content-type'], str):
+            if 'application/json' in response.headers['content-type'].lower():
+                if response.content:
+                    result = response.json()
+                    label = process_results(result)
+                else:
+                    return "Response content missing"
```

Comment:
These return statements without an explicit code default to 200 OK as explained [in the Flask docs](https://flask.palletsprojects.com/en/2.0.x/quickstart/#about-responses). Please change these to return an appropriate 500 code.